src/feedback/data/sagas.js

```diff
@@ -54,3 +54,28 @@ export function* handleMessages(messages, clearExistingMessages, url) {
   }
   return null;
 }
+
+/**
+ * Handle Subscription Errors
+ */
+export function* handleSubscriptionErrors(e, clearExistingMessages) {
+  if (clearExistingMessages) {
+    yield put(clearMessages());
+  }
+  // If this doesn't contain anything we understand, add a fallback error message
+  if (e.errors === undefined && e.fieldErrors === undefined && e.messages === undefined) {
+    yield put(addMessage('fallback-error', null, {}, MESSAGE_TYPES.ERROR));
+  }
+  if (e.errors !== undefined) {
+    for (let i = 0; i < e.errors.length; i++) { // eslint-disable-line no-plusplus
+      const error = e.errors[i];
+      yield put(addMessage(error.code, error.userMessage, error.data, error.messageType));
```

Comment:
Have you tested error handling for when the payment is not processed because we get a decline from Stripe? 
You can use these - https://stripe.com/docs/testing#declined-payments

Reply:
I see! You'll likely need the `TransactionDeclined` in there. You should chat with Chris about the post checkout error messages later because he has a bunch coming in PON-120, but they might just all fall into the fallback error message.